### ISSUE TITLE
Move shader capabilities validation to CreateShaderModule

### ISSUE THREAD

**Issue by jzulauf-lunarg** [KhronosGroup/Vulkan-ValidationLayers#1022]
In fixing a plaform specific crash, VkLayerTest, SubgroupSupportedOperations, it appears that for the case of invalid capabilities some drivers (recent AMD) fail at CreateShaderModule time. The capability test should be moved up to CSM time, implementing ShaderModuleCreateInfo-pCode-01090

The unit test should also be updated to reflect the earlier reporting of the issue.

The workaround for this is implemented in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/1021 for

**Comment by sfricke-samsung** [comment]
So the past few weeks I have started a crusade to improve labeling of SPIR-V Validation for Vulkan from the Spec side of thing. A while back when trying to implement `ShaderModuleCreateInfo-pCode-01090` I realized that capability list is hand edited. I am currently working (internal Vulkan issue) on getting the XML to represent all the Shader capabilities that Vulkan requires so that the list of `struct CapabilityInfo` items can be generated in a maintainable fashion.

With this I also did notice this was not being checked at CreateShaderModule time, so yes, should move the check to.

tl;dr feel free to assign this to me if no one is planning to suddenly tackle this in the short term
